Src/Support/GoogleApis/Apis/[Media]/Upload/ResumableUpload.cs

```diff
@@ -412,7 +391,7 @@ public Task<IUploadProgress> UploadAsync()
 
             try
             {
-                UploadUri = await InitializeUpload(cancellationToken).ConfigureAwait(false);
+                UploadUri = await InitiateSessionAsync(cancellationToken).ConfigureAwait(false);

```

Comment:
Ok, will do.

Actually, I also realize now I misread your comment. You had asked for it to be a local variable and I thought you were asking for it to be a field. No, local variable is not possible. Once the session is initiated, the same instance can be used to to resume the upload if it gets interrupted for any reason. So the URI needs to be cached somewhere.